common/src/main/java/io/pravega/common/util/AbstractBufferView.java

```diff
@@ -0,0 +1,200 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.common.util;
+
+import com.google.common.base.Preconditions;
+import io.pravega.common.hash.HashHelper;
+import java.io.EOFException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+import java.util.Collections;
+import java.util.List;
+import lombok.Getter;
+import lombok.SneakyThrows;
+
+/**
+ * Base implementation of {@link BufferView}, providing common functionality.
+ * Derived classes may override these methods with more efficient implementations tailored to their data types.
+ */
+public abstract class AbstractBufferView implements BufferView {
+    static final BufferView EMPTY = new EmptyBufferView();
+    private static final HashHelper HASH = HashHelper.seededWith(AbstractBufferView.class.getName());
+
+    @Override
+    public int hashCode() {
+        HashHelper.HashBuilder builder = HASH.newBuilder();
+        collect(builder::put);
+        return builder.getAsInt();
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj instanceof BufferView) {
+            return equals((BufferView) obj);
+        }
+        return false;
+    }
+
+    /**
+     * Checks for equality against another {@link BufferView} instance.
+     *
+     * Note: this method provides a general implementation of equality. Derived classes should override this method with
+     * more efficient implementations if available.
+     *
+     * @param other The other {@link BufferView} instance.
+     * @return True if this instance and the other instance have the same contents.
+     */
+    @SneakyThrows(IOException.class)
+    public boolean equals(BufferView other) {
+        int l = getLength();
+        if (l != other.getLength()) {
+            return false;
+        }
+
+        if (l > 0) {
+            BufferView.Reader thisReader = getBufferViewReader();
+            BufferView.Reader otherReader = other.getBufferViewReader();
+            for (int i = 0; i < l; i++) {
+                if (thisReader.readByte() != otherReader.readByte()) {
+                    return false;
+                }
+            }
+        }
+
+        return true;
+    }
+
+    //region AbstractReader
+
+    /**
+     * Base implementation of {@link BufferView.Reader}. Subclasses of {@link AbstractBufferView} should implement their
+     * own {@link BufferView.Reader} instances based on this class.
+     */
+    protected static abstract class AbstractReader implements BufferView.Reader {
+        @Override
+        public int readInt() throws EOFException {
+            return BitConverter.makeInt(readByte(), readByte(), readByte(), readByte());
```

Comment:
I changed `readByte` to return a byte.

I cannot change `makeInt`/`makeLong` to accept bytes since those are inputs from `InputStream` and I also need to account for byte signage.